accepted/2024/[TS]PMUI-Readme-rendering.md

```diff
@@ -0,0 +1,88 @@
+MarkdownPreviewControl = markdownPreview.VisualElement
+```
+
+For packages that are already on the disk we will retrieve the ReadMe from the root folder of the package. We will need to confirm if the current API returns the ReadMe when fetching package details and determine which API we should use to get the data.
+<!-- Explain the proposal in sufficient detail with implementation details, interaction models, and clarification of corner cases. -->
+
+## Drawbacks
+
+This control is not currently using Daytona. There are plans to upgrade it, [Experience 1870733](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1870733), but they have been delayed. 
+
+It's also currently marked as obsolete since the interface has not been finalized. So when an upgrade is made we may have to change how we use the control.
+
+Currently WebView2 controls always render ontop of other controls in the view. [Secnario 25254665](https://microsoft.visualstudio.com/Edge/_workitems/edit/25254665). PM UI needs to be updated to ensure items don't scroll off screen.
+
+## Rationale and alternatives
+By using an existing control we maintain consistency throughout the IDE and can rely on the owner to fix any bugs with the control.
+<!-- Why is this the best design compared to other designs? -->
+<!-- What other designs have been considered and why weren't they chosen? -->
+<!-- What is the impact of not doing this? -->
+
+## Prior Art
+The IMarkdownPreview is currently being used when creating a new pull request inside of Visual Studio.  
+![Alt text](../../meta/resources/ReadMePMUI/PullRequestExperience.png) 
+<!-- What prior art, both good and bad are related to this proposal? -->
+<!-- Do other features exist in other ecosystems and what experience have their community had? -->
+<!-- What lessons from other communities can we learn from? -->
+<!-- Are there any resources that are relevant to this proposal? -->
+
+## Unresolved Questions
+1. Will we show readme in the updates tab?
+1. What do we show if the package has a ReadMe.txt instead of md?
+    * Current suggestion is to focus on ReadMe.md. Once that's complete see how ReadMe.txt would appear when rendered. 
+    * How many packages have a ReadMe.txt vs ReadMe.md?
+1. What do we show if there is no ReadMe defined?
+    - [ ] Hide the tab from the the details pane
+    - [ ] Display the tab with a message saying there is no ReadMe defined.
+1. Where do we get the ReadMe from when it's not on the disk?
+    - Does the current API return the ReadMe information when searching for a package? 
+    - If not is there an API to retrieve the ReadMe information?
+1. Where are the ReadMe files saved in a package? 
+    - There are examples of them being in root folder as well as a _content folder. 
+1. What do we want the UX to be when an exception or error occurs while reading a ReadMe file? 
+1. Do we want the ReadMe to update whenever a new version is selected for the current package?
```

Comment:
@albarry4, do you have thoughts on this? I'm guessing it's possible that the ReadMe may change between versions, so I think it makes sense to render the ReadMe for the current version.

Reply:
I think my (possibly unrealistic) _ideal_ is README should match whatever version is selected. That's how nuget.org works as well when selecting older package versions (you also get the older README). 

If that's too costly, we could instead _always_ show Latest README by default, with some declaration that "we're showing latest README" and a way to load the older, version-specific one.

The reason I'm suggesting we provide options for older and Latest is that I see a use case for both. A newer README may have relevant changes that affect how to use even older versions of the package. At the same time, the older README (historical, yet point-in-time with the package itself) may be more specific to the older APIs and expectations of that older package. I'm not sure if there's a "standard"/"best-practice" for editing your package's README as you update your packages. That knowledge would also help answer this one way or another, but I vote for "latest by default, with flexibility" as a compromise.
 
A reminder of our behavior across tabs:
- Browse tab defaults to Latest Version. Therefore, latest README will be most common.
- Installed tab defaults to Installed Version.

Reply:
Readme's are per id/version, so let's just follow that.

Reply:
If we're focusing on the files that are currently downloaded in the MVP it's likely that when they switch versions the new version won't have be downloaded and the ReadMe will disappear.

Reply:
If a package isn't downloaded, wouldn't we just go fetch the data from the servers anyways? If that's the case, then we'd be able to display it correct?